<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="robot" default="machines"/>
    <arg name="debug" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="pause" default="true"/>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find my_legged_robots_sims)/worlds/low_gravity.world"/>
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg pause)"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="headless" value="$(arg headless)"/>
        <env name="GAZEBO_MODEL_PATH" value="$(find my_legged_robots_sims)/models:$(optenv GAZEBO_MODEL_PATH)"/>
    </include>
    
    <include file="$(find my_legged_robots_sims)/launch/spawn_monoped.launch">
    </include>
    <include file="$(find my_legged_robots_sims)/launch/monoped_control.launch"/>
    <node pkg="my_legged_robots_sims" type="joint_publisher.py" name="joint_publisher_node" output="screen"/>
    <!-- <node pkg="my_legged_robots_sims" type="towr_ros_interface.py" name="towr_ros_interface_node" output="screen"/> -->
</launch>